paint/support/procedural/src/construct_runtime/mod.rs

```diff
@@ -0,0 +1,392 @@
+            type NodeBlock = #node_block;
+        }
+        impl #scrate::sr_primitives::traits::GetRuntimeBlockType for #name {
+            type RuntimeBlock = #block;
+        }
+
+        #outer_event
+
+        #outer_origin
+
+        #all_modules
+
+        #dispatch
+
+        #metadata
+
+        #outer_config
+
+        #inherent
+
+        #validate_unsigned
+    )
+    .into())
+}
+
+fn decl_validate_unsigned<'a>(
+    runtime: &'a Ident,
+    module_declarations: impl Iterator<Item = &'a ModuleDeclaration>,
+    scrate: &'a TokenStream2,
+) -> TokenStream2 {
+    let modules_tokens = module_declarations
+        .filter(|module_declaration| {
+            module_declaration
+                .module_parts()
+                .into_iter()
+                .any(|part| part.name == "ValidateUnsigned")
+        })
+        .map(|module_declaration| &module_declaration.name);
+    quote!(
+        #scrate::impl_outer_validate_unsigned!(
+            impl ValidateUnsigned for #runtime {
+                #( #modules_tokens )*
+            }
+        );
+    )
+}
+
+fn decl_outer_inherent<'a>(
+    block: &'a syn::TypePath,
+    unchecked_extrinsic: &'a syn::TypePath,
+    module_declarations: impl Iterator<Item = &'a ModuleDeclaration>,
+    scrate: &'a TokenStream2,
+) -> TokenStream2 {
+    let modules_tokens = module_declarations.filter_map(|module_declaration| {
+        let maybe_config_part = module_declaration
+            .module_parts()
+            .into_iter()
+            .filter(|part| part.name == "Inherent")
+            .next();
+        maybe_config_part.map(|config_part| {
+            let arg = config_part
+                .args
+                .as_ref()
+                .and_then(|parens| parens.content.inner.iter().next())
+                .unwrap_or(&module_declaration.name);
+            let name = &module_declaration.name;
+            quote!(#name : #arg,)
+        })
+    });
+    quote!(
+        #scrate::impl_outer_inherent!(
+            impl Inherents where Block = #block, UncheckedExtrinsic = #unchecked_extrinsic {
+                #(#modules_tokens)*
+            }
+        );
+    )
+}
+
+fn decl_outer_config<'a>(
+    runtime: &'a Ident,
+    module_declarations: impl Iterator<Item = &'a ModuleDeclaration>,
+    scrate: &'a TokenStream2,
+) -> TokenStream2 {
+    let modules_tokens = module_declarations
+        .filter_map(|module_declaration| {
+            let generics: Vec<_> = module_declaration
+                .module_parts()
+                .into_iter()
+                .filter(|part| part.name == "Config")
+                .map(|part| part.generics)
+                .collect();
+            if generics.len() == 0 {
+                None
+            } else {
+                let transformed_generics: Vec<_> = generics[0]
+                    .params
+                    .iter()
+                    .map(|param| quote!(<#param>))
```

Comment:
This can be just a `map()` on the value from above.

Reply:
I'm not sure what you mean here.  The thing is that in case of 2 params decl macros accept something like `<T> <I>` instead of `<T, I>` - that's what this code does.

Reply:
Where do we accept `<T><I>`?

And what I wanted is, you should not call `collect` in line 188. Use `find` and then you just get an `Option` that you can map directly.

Reply:
https://github.com/paritytech/substrate
> Where do we accept <T><I>?

/blob/ba72d757254efcda1375bd447212e5149cb432a8/primitives/sr-primitives/src/lib.rs#L523-L529

As far as I can see it accepts `<T>, <I>` vs `<T, I>`

Reply:
> And what I wanted is, you should not call collect in line 188
I see, I made the fix to make the code shorter